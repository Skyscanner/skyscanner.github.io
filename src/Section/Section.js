/**
 * Copyright 2021 Skyscanner Limited.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import ReactMarkdown from 'react-markdown';
import BpkText, {
  TEXT_STYLES,
} from '@skyscanner/backpack-web/bpk-component-text';
import { cssModules } from '@skyscanner/backpack-web/bpk-react-utils';

import STYLES from './Section.scss';

const c = cssModules(STYLES);

const Section = (props) => {
  // eslint-disable-next-line react/prop-types
  const { content, title } = props;
  return (
    <div className={c('Section')}>
      {title && (
        <div>
          <section className={c('Section__heading')}>
            <div className={c('Section__heading-title')}>
              <BpkText tagName="h2" textStyle={TEXT_STYLES.subheading}>
                {title}
              </BpkText>
            </div>
          </section>
          <hr className={c('Section__divider')} />
        </div>
      )}
      <section className={c('Section__main')}>
        <ReactMarkdown>{content}</ReactMarkdown>
      </section>
    </div>
  );
};

Section.defaultProps = {
  title: null,
};

export default Section;
