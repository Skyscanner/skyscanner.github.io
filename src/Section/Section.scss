/**
 * Copyright 2021 Skyscanner Limited.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '../tokens.scss';

.Section {
  margin-bottom: 1.5rem;
  /* stylelint-disable-next-line scale-unlimited/declaration-strict-value */
  line-height: 1.3rem; /* stylelint-disable-line backpack/use-typography-styles */
  text-align: left;

  &__divider {
    margin-top: bpk-spacing-base();
    margin-bottom: bpk-spacing-base();
    border: 0;
    border-top: $bpk-one-pixel-rem solid $bpk-surface-highlight-day;
  }

  &__heading-title {
    display: flex;
    margin-bottom: bpk-spacing-md();
    justify-content: space-between;
  }

  &__heading {
    margin-bottom: bpk-spacing-base();
  }

  &__main {
    position: relative;
    width: 100%;

    // Style links correctly.
    a {
      @include bpk-link;
    }
  }
}
