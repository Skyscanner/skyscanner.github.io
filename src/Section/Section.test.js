/**
 * Copyright 2021 Skyscanner Limited.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { render } from '@testing-library/react';

import Section from './Section';

describe('Section tests', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Section title="Section title" content="Section content" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no title', () => {
    const { asFragment } = render(<Section content="Section content" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
