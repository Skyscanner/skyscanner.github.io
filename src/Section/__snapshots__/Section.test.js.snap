// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section tests renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Section"
  >
    <div>
      <section
        class="Section__heading"
      >
        <div
          class="Section__heading-title"
        >
          <h2
            class="bpk-text bpk-text--subheading"
          >
            Section title
          </h2>
        </div>
      </section>
      <hr
        class="Section__divider"
      />
    </div>
    <section
      class="Section__main"
    >
      <p>
        Section content
      </p>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Section tests renders correctly with no title 1`] = `
<DocumentFragment>
  <div
    class="Section"
  >
    <section
      class="Section__main"
    >
      <p>
        Section content
      </p>
    </section>
  </div>
</DocumentFragment>
`;
